File: app/src/main/java/com/eusebeia/faithbuilders/SpeakersFragment.java
package com.eusebeia.faithbuilders;


import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.TextView;

/**
 * A simple {@link Fragment} subclass.
 */
public class SpeakersFragment extends Fragment {

    public SpeakersFragment() {
        // Required empty public constructor
    }

    @Override
    public void onResume() {
        super.onResume();
        // Set title
        getActivity().setTitle(R.string.label_speakers);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View v = inflater.inflate(R.layout.fragment_speakers, container, false);
        populateListView(v);
        registerClickCallback(v);

        // Inflate the layout for this fragment
        return v;
    }

    private void populateListView(View v) {
        ArrayAdapter<Speaker> adapter = new MyListAdapter();
        ListView list = (ListView) v.findViewById(R.id.ListViewSpeakers);
        list.setAdapter(adapter);
    }

    private void registerClickCallback(final View v) {
        ListView list = (ListView) v.findViewById(R.id.ListViewSpeakers);
        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View viewClicked, int position, long id) {

                DialogFragment newFragment = new BiographyFragment();
                newFragment.show(getFragmentManager(), "biography");
                Bundle bundle = new Bundle();
                bundle.putInt("SpeakerIndex", position);
                newFragment.setArguments(bundle);
            }
        });
    }

    private class MyListAdapter extends ArrayAdapter<Speaker> {
        public MyListAdapter() {
            super(getActivity(), R.layout.speaker_list_item, MainActivity.SpeakerList);
        }

        @Override
        public View getView(int position, View convertView, ViewGroup parent) {
            View itemView = convertView;
            if (itemView == null)
                itemView = getActivity().getLayoutInflater().inflate(R.layout.speaker_list_item, null);

            // Find the speaker to work with.
            Speaker currentSpeaker = MainActivity.SpeakerList.get(position);

            // Fill the view.
            // Photo
            ImageView imageView = (ImageView) itemView.findViewById(R.id.itemIcon);
            imageView.setImageResource(currentSpeaker.getPhoto());

            // Name
            TextView textView = (TextView) itemView.findViewById(R.id.itemName);
            textView.setText(currentSpeaker.getName());

            // Location
            textView = (TextView) itemView.findViewById(R.id.itemLocation);
            textView.setText(currentSpeaker.getLocation());

            return itemView;
        }
    }
}

File: app/src/main/java/com/eusebeia/faithbuilders/SpeakersFragment.kt
package com.eusebeia.faithbuilders


import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * A simple [Fragment] subclass.
 */
class SpeakersFragment : Fragment(), SpeakersAdapter.ClickHandler {
	
	override fun onResume() {
		super.onResume()
		// Set title
		activity?.setTitle(R.string.label_speakers)
	}
	
	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
		val view = inflater.inflate(R.layout.fragment_speakers, container, false)
		populateListView(view)
		return view
	}
	
	override fun onItemClicked(position: Int) {
		val newFragment = BiographyFragment()
		newFragment.show(fragmentManager, "biography")
		val bundle = Bundle()
		bundle.putInt("SpeakerIndex", position)
		newFragment.arguments = bundle
	}
	
	private fun populateListView(view: View) {
		val recyclerView = view.findViewById<RecyclerView>(R.id.recyclerViewSpeakers)
		val layoutManager = LinearLayoutManager(view.context)
		recyclerView.layoutManager = layoutManager
		recyclerView.adapter = SpeakersAdapter(this)
		recyclerView.addItemDecoration(DividerItemDecoration(recyclerView.context, layoutManager.orientation))
	}
}
